x-pack/plugins/observability/public/components/shared/page_template/nav_name_with_beta_badge.tsx

```diff
@@ -4,45 +4,31 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { EuiBetaBadge } from '@elastic/eui';
-import type { IconType } from '@elastic/eui/src/components/icon/icon';
-import { i18n } from '@kbn/i18n';
 import React from 'react';
-import styled from 'styled-components';
+import { i18n } from '@kbn/i18n';
+import { EuiBetaBadge, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
 
 interface Props {
-  iconType: IconType;
-  label: string;
+  label?: string;
 }
 
-const LabelContainer = styled.span`
-  max-width: 72%;
-  float: left;
-  &:hover,
-  &:focus-within,
-  &:focus {
-    text-decoration: underline;
-  }
-`;
-
-const StyledBetaBadge = styled(EuiBetaBadge)`
-  margin-left: 5px;
-  margin-bottom: -6px;
-`;
-
-export function NavNameWithBetaBadge({ label, iconType }: Props) {
+export function NavNameWithBetaBadge({ label }: Props) {
```

Comment:
So my idea is to keep a single component and to have for example 2 more optional props: `{ label, iconType, isTechnicalPreview }`